### PR TITLE
Add types and decorator for TypeScript

### LINKED ISSUE #16: Not working with typescript
This does not appear to work... really excited to use it though. 

I added the plugin, copy and pasted the code (minus the configuration)
Added the `jsonld()` to my index.vue file within pages. 

Nothing happened. Added to default.vue within layouts also, got nothing. 

Looked at the code and reviewed the /examples folder. I saw there was a plugin line within `nuxt.config.js` which was missing from the docs (is it needed?). Added that, nothing.

I changed the extension from `.js` to `.ts` as I use a typescript project. Nothing.

I assume this is suppose to add this to the head of the site. 

I have viewed the source on the output, both using `nuxt` and `nuxt generate` with all the different combinations and attempts. Nothing.

I can search the souirce, and I do see a JavaScript file appears to show that i was compiled somewhere along the way. 

```
(window.webpackJsonp = window.webpackJsonp || []).push([[2], {
    295: function(e, t, n) {
        "use strict";
        n.r(t);
        n(41);
        var c = n(23),
            o = n(40),
            r = n(26),
            l = n(24),
            f = n(25),
            h = n(5),
            d = n(2),
            w = function(e, t, n, desc) {
                var c,
                    o = arguments.length,
                    r = o < 3 ? t : null === desc ? desc = Object.getOwnPropertyDescriptor(t, n) : desc;
                if ("object" === ("undefined" == typeof Reflect ? "undefined" : Object(h.a)(Reflect)) && "function" == typeof Reflect.decorate)
                    r = Reflect.decorate(e, t, n, desc);
                else
                    for (var i = e.length - 1; i >= 0; i--)
                        (c = e[i]) && (r = (o < 3 ? c(r) : o > 3 ? c(t, n, r) : c(t, n)) || r);
                return o > 3 && r && Object.defineProperty(t, n, r), r
            },
            j = function(e) {
                function t() {
                    return Object(c.a)(this, t), Object(r.a)(this, Object(l.a)(t).apply(this, arguments))
                }
                return Object(f.a)(t, e), Object(o.a)(t, [{
                    key: "jsonld",
                    value: function() {
                        return {
                            "@context": "https://schema.org",
                            "@type": "Organization",
                            url: "https://www.example.com",
                            logo: "https://www.example.com/logo.png",
                            contactPoint: [{
                                "@type": "ContactPoint",
                                telephone: "+1-401-555-1212",
                                contactType: "customer service"
                            }]
                        }
                    }
                }]), t
            }(d.Vue),
            O = j = w([Object(d.Component)({})], j),
            y = n(12),
            component = Object(y.a)(O, function() {
                this.$createElement;
                this._self._c;
                return this._m(0)
            }, [function() {
                var e = this.$createElement,
                    t = this._self._c || e;
                return t("section", [t("h1", {
                    staticClass: "header"
                }, [this._v("Content")])])
            }], !1, null, "e1b7e7d8", null);
        t.default = component.exports
    }
}]);
```

### Plugin File
```
import Vue from "vue";
import NuxtJsonld from "nuxt-jsonld";

Vue.use(NuxtJsonld);

```

### PR DESCRIPTION
**Pull request by ymmooot**
fix #16

JSON-LD was not dumped when using `vue-property-decorator` because `jsonld` method has to be in `options`, not in `methods`.
I created `@Jsonld` decorator to do it.

### CODE REVIEW

File: src/decorator.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export default ({ options }) => {
```
Comment (TKNGUE) [inline src/decorator.js]: ここななるほど

File: package.json [hunk 2]
```diff
@@ -12,20 +12,23 @@
...
     "release": "semantic-release",
-    "lint": "eslint . --ignore-pattern .gitignore",
-    "lint:fix": "eslint . --ignore-pattern .gitignore --fix"
+    "lint": "eslint src",
+    "lint:fix": "eslint --fix src"
   },
   "files": [
-    "lib"
+    "lib",
+    "types/*.d.ts"
```
Comment (RyoNkmr) [inline package.json]: [NITS]
```suggestion
    "types"
```

File: src/index.js [hunk 3]
```diff
@@ -36,7 +37,9 @@ const mergeStrategy = (toVal, fromVal) => {
   };
 };
 
-module.exports = {
+export const Jsonld = decorator;
+
+export default {
```
Comment (RyoNkmr) [inline src/index.js]: 👍

**Pull request merged**